File: to-do-rest/internal/models/todo.go
package models

import "time"

type Todo struct {
	ID        int    `json:"id"`
	ListID	int	`json:"listId"`
	Name      string `json:"name"`
	Completed bool   `json:"completed"`
	CreatedAt time.Time 
}

type TodoList struct {
	ID    int    `json:"id"`
	Name  string `json:"name"`
	Todos []Todo `json:"todos,omitempty"`
}

File: to-do-grpc/web/src/app/model/todo.ts
export interface TodoList {
    ID: number,
    UserID: number,
    Name?: string,
    CreatedAt?: string 
    Todos?: Todo[]
}

export interface CreateTodoListRequest {
    UserID: number,
    Name: string
}

export interface DeleteTodoListRequest {
    UserID: number,
    ID: number
}

export interface Todo {
    ID: number,
    ListID: number,
    Name: string, 
    Completed: boolean,
    CretedAt: string
}

export interface CreateTodoRequest {
    UserID: number,
    ListID: number,
    Name: string,
}

export interface UpdateTodoRequest {
    ID: number,
    UserID: number,
    ListID: number,
    Name: string,
    Completed: boolean,
}

export interface DeleteTodoRequest {
    ID: number,
    UserID: number,
    ListID: number
}

